### PR TITLE
Social app

### PR DESCRIPTION
**Pull request by franklovefrank**
# Description

example SSX-powered app that has a universal post feed, and lets signed-in users add a post. This version is configured to run locally and does not persist state

# Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Diligence Checklist

(Please delete options that are not relevant)

- [X] This change requires a documentation update
- [X] I have performed a self-review of my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: examples/ssx-test-social/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# ssx-test-dapp
```
Comment (Juliano1612) [inline examples/ssx-test-social/README.md]:
```suggestion
# ssx-test-social
```

File: examples/ssx-test-social/src/interfaces/iPost.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { Address } from "cluster";
+
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/interfaces/iPost.ts]:
```suggestion
```

File: examples/ssx-test-social/src/interfaces/iPostContext.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { SSX } from '@spruceid/ssx'
+import Post from './iPost'
+
+export default interface PostContextProps {
+  ssxProvider: SSX | null,
+  setSSXProvider: React.Dispatch<React.SetStateAction<SSX | null>>,
+  posts: Post[],
+  setPosts: React.Dispatch<React.SetStateAction<Post[]>>
+
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/interfaces/iPostContext.ts]:
```suggestion
```

File: examples/ssx-test-social/src/components/AccountInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Card, CardContent, Typography } from "@material-ui/core";
+
+function AccountInfo({ address, delegator }: { address: string, delegator?: string }) {
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/components/AccountInfo.tsx]:
```suggestion
function AccountInfo({ address }: { address: string }) {
```

File: examples/ssx-test-social/src/components/NewPost.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import { useContext, useState } from "react";
+import { TextField, Button, InputAdornment, IconButton } from '@material-ui/core';
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/components/NewPost.tsx]:
```suggestion
import { TextField, InputAdornment, IconButton } from '@material-ui/core';
```

File: examples/ssx-test-social/src/components/SinglePost.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/components/SinglePost.tsx]:
```suggestion
```

File: examples/ssx-test-social/src/components/Feed.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import { useContext, useEffect, useState } from 'react';
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/components/Feed.tsx]:
```suggestion
import { useContext, useEffect } from 'react';
```

File: examples/ssx-test-social/src/ssx.config.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    providerOptions: {
+      walletconnect: {
+        package: WalletConnectProvider,
+      },
+    },
+  }).connect();
+
+  return { 
+    enableDaoLogin: true,
+    //resolveEns: true,
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/ssx.config.js]:
```suggestion
```

File: examples/ssx-test-social/hardhat.config.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
+/**
```
Comment (Juliano1612) [inline examples/ssx-test-social/hardhat.config.ts]: Do we need to use Hardhat on this example? I removed the dependency and the hardhat.config.ts locally and the app is still working.

File: examples/ssx-test-social/package.json [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  },
+  "devDependencies": {
+    "@testing-library/jest-dom": "^5.16.4",
+    "@testing-library/react": "^13.3.0",
+    "@testing-library/user-event": "^13.5.0",
+    "@types/jest": "^27.5.2",
+    "@types/node": "^16.11.42",
+    "@types/react": "^18.0.14",
+    "@types/react-dom": "^18.0.5",
+    "hardhat": "^2.12.6",
```
Comment (Juliano1612) [inline examples/ssx-test-social/package.json]: Same here

File: examples/ssx-test-social/.env.example [hunk 11]
```diff
@@ -0,0 +1,7 @@
+REACT_APP_SSX_METRICS_SERVER=
+REACT_APP_SSX_DAO_LOGIN=
+REACT_APP_INFURA_ID=
+API_URL = "https://eth-goerli.g.alchemy.com/v2/<your-api-key>"
+API_KEY = "<your-api-key>"
+PRIVATE_KEY = "<your-metamask-private-key>"
+CONTRACT_ADDRESS = "0x<your contract address>"
```
Comment (Juliano1612) [inline examples/ssx-test-social/.env.example]:
```suggestion
```

Same about Hardhat here

**Review by Juliano1612 — CHANGES_REQUESTED** [review]

File: examples/ssx-test-social/src/App.test.tsx [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (w4ll3) [inline examples/ssx-test-social/src/App.test.tsx]: This can be deleted

File: examples/ssx-test-social/src/components/Feed.tsx [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    return (
+      <div style={{ display: 'flex', flexDirection: 'column-reverse', marginTop: '2rem' }}>
+        {posts.map((post, index) => (
+          <SinglePost key={index} user={post.user} postText={post.postText} />
+        ))}
+      </div>
+    );
+  }
+
+  if (posts && posts.length > 0) {
```
Comment (w4ll3) [inline examples/ssx-test-social/src/components/Feed.tsx]:
```suggestion
  if (posts?.length > 0) {
```

File: examples/ssx-test-social/src/components/Feed.tsx [hunk 14]
```diff
@@ -0,0 +1,32 @@
+import { useContext, useEffect, useState } from 'react';
+import PostContextProps from '../interfaces/iPostContext';
+import React from 'react';
+import SinglePost from './SinglePost';
+
+
+function Feed({ postContext }: { postContext: React.Context<PostContextProps> }) {
+  const { posts } = useContext(postContext);
+
+  useEffect(() => {
```
Comment (w4ll3) [inline examples/ssx-test-social/src/components/Feed.tsx]: do you really need this hook?

File: examples/ssx-test-social/src/components/NewPost.tsx [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+import Post from "../interfaces/iPost";
+import SendIcon from '@mui/icons-material/Send';
+
+function NewPost({ postContext }: { postContext: React.Context<PostContextProps> }) {
+  const { ssxProvider, posts, setPosts } = useContext(postContext);
+  const [text, setText] = useState('');
+
+  const handleSubmit = async (event: any) => {
+    event.preventDefault();
+    const address = (ssxProvider instanceof SSX && typeof ssxProvider.address() !== 'undefined')
```
Comment (w4ll3) [inline examples/ssx-test-social/src/components/NewPost.tsx:14]: can't this all be resumed to `ssxProvider?.address()`?

File: examples/ssx-test-social/package.json [hunk 16]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "ssx-test-social",
+  "version": "0.0.1",
+  "private": true,
+  "dependencies": {
+    "@material-ui/core": "^4.12.4",
+    "@mui/icons-material": "^5.11.0",
+    "@mui/material": "^5.11.6",   
+     "@emotion/react": "^11.10.5",
```
Comment (Juliano1612) [inline examples/ssx-test-social/package.json]:
```suggestion
    "@emotion/react": "^11.10.5",
```

File: examples/ssx-test-social/src/components/Feed.tsx [hunk 17]
```diff
@@ -0,0 +1,29 @@
+import { useContext} from 'react';
```
Comment (Juliano1612) [inline examples/ssx-test-social/src/components/Feed.tsx]:
```suggestion
import { useContext } from 'react';
```

**Review by Juliano1612 — APPROVED** [review]
Looks good to me!

**Review by Juliano1612 — APPROVED** [review]

**Review by w4ll3 — APPROVED** [review]

File: examples/ssx-test-social/README.md [hunk 18]
```diff
@@ -0,0 +1,34 @@
+# ssx-test-social
+
+This is a simple social media that demonstrates the functionality of SSX. It is configured to run locally and does not persist state between instances.
```
Comment (obstropolos) [inline examples/ssx-test-social/README.md:3]: "social media" -> "social media dapp"

**Comment by obstropolos** [comment]
Closing - will be consolidated under a repository - `ssx-examples`

**Pull request closed without merging**